component/src/main/java/me/androidbox/component/general/PasswordEntry.kt

```diff
@@ -0,0 +1,112 @@
+package me.androidbox.component.general
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import me.androidbox.component.R
+import me.androidbox.component.ui.theme.*
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun PasswordEntry(
+    modifier: Modifier = Modifier,
+    passwordValue: String,
+    placeholderText: String,
+    isPasswordVisible: MutableState<Boolean> = mutableStateOf(false),
+    onPasswordChange: (String) -> Unit,
+) {
+    OutlinedTextField(
+        modifier = modifier
+            .fillMaxWidth()
```

Comment:
I'd not force the text field to fill the whole width in this composable. Imagine, you'd want to show 2 text fields next to each other. That wouldn't be possible with your implementation, since both would try to fill the whole width and the left one would push the other one out of the screen.
Where a composable is positioned and how large it is should be decided outside that composable at the screen level 👍